internal/daemon/worker/controller_connection.go

```diff
@@ -77,6 +78,14 @@ func (w *Worker) controllerDialerFunc() func(context.Context, string) (net.Conn,
 			conn, err = w.v1KmsAuthDialFn(ctx, addr)
 		default:
 			conn, err = protocol.Dial(ctx, w.WorkerAuthStorage, addr, nodeenrollment.WithWrapper(w.conf.WorkerAuthStorageKms))
+			// No error and a valid connection means the WorkerAuthRegistrationRequest was populated
+			// We can remove the stored workerAuthRequest file
+			if err == nil && conn != nil {
+				if w.WorkerAuthStorage.BaseDir() != "" {
+					workerAuthReqFilePath := fmt.Sprintf("%s/%s", w.WorkerAuthStorage.BaseDir(), base.WorkerAuthReqFile)
```

Comment:
Same comment about `filepath.Join`. Also for the os.Remove command I'd so `_ = os.Remove...` to make it clear that this isn't forgotten error handling but we explicitly want to ignore a failure since it isn't really a failure of the system overall (and add a comment about it)